### PR TITLE
feat(testing): convert CI testing to use Makefile actions

### PR DESCRIPTION
**Pull request by muncus**
This updates all current github actions workflows that called `npm test` to use the Makefile task instead.

It also updates the templates, so newly-created github actions should also use this method.

To ensure that tests are run, path matches were added to all workflows that reuse the `test.yaml` workflow, so changes to the underlying workflow will also trigger the tests.

### CODE REVIEW

File: .github/workflows/ai-platform-snippets.yaml [hunk 1]
```diff
@@ -64,30 +61,26 @@ jobs:
...
       if: github.event.action == 'schedule'
       run: |
         echo "MOCHA_REPORTER_SUITENAME=ai-platform-snippets" >> $GITHUB_ENV
         echo "MOCHA_REPORTER_OUTPUT=${{github.run_id}}_sponge_log.xml" >> $GITHUB_ENV
         echo "MOCHA_REPORTER=xunit" >> $GITHUB_ENV
-    - run: npm test
+    - name: Run Tests
+      run: make test dir=ai-platform/snippets
       env:
+        GOOGLE_SAMPLES_PROJECT: "nodejs-docs-samples-tests"
```
Comment (pattishin) [inline .github/workflows/ai-platform-snippets.yaml:81]:
```suggestion
        GOOGLE_SAMPLES_PROJECT: "long-door-651"
```

File: .github/workflows/dialogflow-cx.yaml [hunk 2]
```diff
@@ -67,28 +61,24 @@ jobs:
...
       if: github.event.action == 'schedule'
       run: |
         echo "MOCHA_REPORTER_SUITENAME=dialogflow-cx" >> $GITHUB_ENV
         echo "MOCHA_REPORTER_OUTPUT=${{github.run_id}}_sponge_log.xml" >> $GITHUB_ENV
         echo "MOCHA_REPORTER=xunit" >> $GITHUB_ENV
-    - run: npm test
+    - name: Run Tests
+      run: make test dir=dialogflow-cx
       env:
+        GOOGLE_SAMPLES_PROJECT: "nodejs-docs-samples-tests"
```
Comment (pattishin) [inline .github/workflows/dialogflow-cx.yaml:81]:
```suggestion
         GOOGLE_SAMPLES_PROJECT: "long-door-651"
```

File: .github/workflows/functions-slack.yaml [hunk 3]
```diff
@@ -64,28 +61,24 @@ jobs:
...
       if: github.event.action == 'schedule'
       run: |
         echo "MOCHA_REPORTER_SUITENAME=functions-slack" >> $GITHUB_ENV
         echo "MOCHA_REPORTER_OUTPUT=${{github.run_id}}_sponge_log.xml" >> $GITHUB_ENV
         echo "MOCHA_REPORTER=xunit" >> $GITHUB_ENV
-    - run: npm test
+    - name: Run Tests
+      run: make test dir=functions/slack
       env:
+        GOOGLE_SAMPLES_PROJECT: "nodejs-docs-samples-tests"
```
Comment (pattishin) [inline .github/workflows/functions-slack.yaml:81]:
```suggestion
        GOOGLE_SAMPLES_PROJECT: "long-door-651"
```

File: .github/workflows/utils/ci-secrets.yaml.njk [hunk 4]
```diff
@@ -65,28 +62,24 @@ jobs:
...
       if: github.event.action == 'schedule'
       run: |
         echo "MOCHA_REPORTER_SUITENAME={{ name }}" >> $GITHUB_ENV
         echo "MOCHA_REPORTER_OUTPUT=${% raw %}{{github.run_id}}{% endraw %}_sponge_log.xml" >> $GITHUB_ENV
         echo "MOCHA_REPORTER=xunit" >> $GITHUB_ENV
-    - run: npm test
+    - name: Run Tests
+      run: make test dir={{ path }}
       env:
+        GOOGLE_SAMPLES_PROJECT: "nodejs-docs-samples-tests"
```
Comment (pattishin) [inline .github/workflows/utils/ci-secrets.yaml.njk:82]:
```suggestion
        GOOGLE_SAMPLES_PROJECT: "long-door-651"
```

**Review by pattishin — CHANGES_REQUESTED** [review]
Ty Marc! Added small suggestions for GCP project environment.

**Review by holtskinner — APPROVED** [review]

**Review by pattishin — APPROVED** [review]
👍🏼 ty!

**Pull request merged**